mqtt/mqtt-broker-tests-util/src/client.rs

```diff
@@ -240,27 +239,40 @@ where
         let (sub_sender, sub_receiver) = mpsc::unbounded_channel();
         let (conn_sender, conn_receiver) = mpsc::unbounded_channel();
 
-        let (termination_handle, tx) = oneshot::channel::<()>();
+        let (termination_handle, termination_receiver) = mpsc::channel::<()>(1);
```

Comment:
why not oneshot?

Reply:
oneshot is not a stream, and select needs a fused stream.... this is a negligible downside of this approach.

Reply:
As discussed offline, it looks like I could use FutureExt and fuse oneshot future